libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -739,14 +622,47 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
         *pOperationReference = pSubscriptionOperation;
     }
 
-    /* Schedule the subscription operation for network transmission. */
-    status = _IotMqtt_ScheduleOperation( pSubscriptionOperation,
-                                         _IotMqtt_ProcessSend,
-                                         0 );
+    if( pSubscriptionOperation->u.operation.status == IOT_MQTT_STATUS_PENDING )
+    {
+        if( operation == IOT_MQTT_SUBSCRIBE )
+        {
+            /* Calling SUBSCRIBE wrapper to send SUBSCRIBE packet on the network using MQTT LTS SUBSCRIBE API. */
+            status = _IotMqtt_managedSubscribe( mqttConnection,
+                                                pSubscriptionOperation,
+                                                pSubscriptionList,
+                                                subscriptionCount );
+        }
 
-    if( status != IOT_MQTT_SUCCESS )
+        if( operation == IOT_MQTT_UNSUBSCRIBE )
+        {
+            /* Calling UNSUBSCRIBE wrapper to send UNSUBSCRIBE packet on the network using MQTT LTS UNSUBSCRIBE API. */
+            status = _IotMqtt_managedUnsubscribe( mqttConnection,
+                                                  pSubscriptionOperation,
+                                                  pSubscriptionList,
+                                                  subscriptionCount );
+        }
+    }
+
+    if( status == IOT_MQTT_SUCCESS )
+    {
+        contextIndex = _IotMqtt_getContextIndexFromConnection( mqttConnection );
+
+        xSemaphoreTakeRecursive( ( SemaphoreHandle_t ) &( connToContext[ contextIndex ].referencesMutex ), portMAX_DELAY );
+
+        if( contextIndex >= 0 )
```

Comment:
Please add more comments in the code about what is the summary of the operations here.